src/fennel/macros.fnl

```diff
@@ -152,10 +152,23 @@ returns
 (fn partial* [f ...]
   "Returns a function with all arguments partially applied to f."
   (assert f "expected a function to partially apply")
-  (let [body (list f ...)]
-    (table.insert body _VARARG)
-    `(fn [,_VARARG]
-       ,body)))
+  (let [bindings []
+        args []]
+    (each [_ arg (ipairs [...])]
+      (if (or (= :number (type arg))
```

Comment:
I think this can actually be shortened to `(not= :table (type arg))` since that covers symbols as well as other tables; does that sound right to you?

Reply:
I think it would have to be `(or (not= :table (type arg)) (= 'nil arg))` since it looks like nil is parsed as a symbol, but I'm good with that overall. I assume it isn't possible for the parser to return userdata? That's the only other exception I can think of.

Reply:
Oh true; I forgot that nil is still a symbol at this point.

Userdata and coroutines are the other types but yeah, they are not valid inside an AST.